library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -1299,73 +1331,74 @@ def GetDevTypeDescription(type, longdescr)
     # @return dumped settings (later acceptable by Import())
     def Export(devregex)
       devs = Filter(@Devices, devregex)
-      Builtins.y2debug("Devs=%1", devs)
-      Convert.convert(devs, from: "map", to: "map <string, map>")
+      log.debug("Devs=#{devs}")
+      devs
     end
 
     # Were the devices changed?
     # @return true if modified
     def Modified(devregex)
       devs = Filter(@Devices, devregex)
       original_devs = Filter(@OriginalDevices, devregex)
-      Builtins.y2debug("OriginalDevs=%1", original_devs)
-      Builtins.y2debug("Devs=%1", devs)
+      log.debug("OriginalDevs=#{original_devs}")
+      log.debug("Devs=#{devs}")
       devs == original_devs
     end
 
+    # It returns an array with the next N devices available
+    # for the given device type.
+    #
+    # @example GetFreeDevices("eth", 2) -> [1, 2]
+    #
+    # @param [String] type device type
+    # @param [String] num number of free devices to return
+    # @return [Array] of free devices for given type
     def GetFreeDevices(type, num)
-      Builtins.y2debug("Devices=%1", @Devices)
-      Builtins.y2debug("type,num=%1,%2", type, num)
-      Builtins.y2debug("Devices[%1]=%2", type, Ops.get(@Devices, type, {}))
+      log.debug("Devices=#{@Devices}")
+      log.debug("type,num=#{type},#{num}")
+      log.debug("Devices[#{type}]=#{@Devices[type]}")
 
-      curdevs = []
-      Builtins.foreach(
-        Convert.convert(
-          Map.Keys(Ops.get(@Devices, type, {})),
-          from: "list",
-          to:   "list <string>"
-        )
-      ) do |dev|
-        dev = device_num(dev) if Builtins.issubstring(dev, type)
-        curdevs = Builtins.add(curdevs, dev)
-      end
+      curdevs = @Devices.fetch(type, {}).keys
+      curdevs.map! { |d| d.include?(type) ? device_num(d) : d }
```

Comment:
please drop call to ```device_num``` it is obsolete function and assumes that device names are in form ```<type><num>``` Since we support e.g. ```cool_name```s for ```eth``` device it makes no sense to use such function and we should replace all occurrences by something more reasonable.